src/app/locationForm/common/Form.jsx

```diff
@@ -15,8 +15,18 @@ class Form extends Component {
     }
   }
 
+  isEditing(props){
+    return props.value === null ||  
+      ( typeof props.value === 'object' ? 
+          Object.keys(props.value).some( key => !props.value[key] ) :
+          !props.value
+      );
+  }
+
   componentWillReceiveProps(props){
-    this.setState({ isEditing: !props.value });
+    this.setState({ 
+      isEditing: this.isEditing(props)
```

Comment:
Same comment about componentWillReceiveProps noted elsewhere.

Reply:
I think this is an appropriate use of componentWillReceiveProps. I don't think we are likely to lose data here, because we will not render the form UI until the data is loaded. But we need to initialize the state.isEditing when the data is loaded, and it would make sense to do this in componentWillReceiveProps. After state.isEditing is set, then the form will be displayed, and the user can edit data.

Reply:
Yeah, you're right. I guess we have to support changing the value after the component mounts, since it may need the time to load...

It almost feels like the user-entered state should be separate from the prop, and render should decide which of the two to use: rendering with the value from the prop as long as the user-entered state doesn't exist yet, and then switching to rendering with the value from state.
But, that sounds more complicated.

Another option would be to remove the component state entirely, and manage it all in Redux state, which is what I've seen done in a lot of apps. That way the component can always just render the props, and on change dispatch an action. Submitting, at that point, is just a matter of calling the API with the values from Redux state.
But that, again, sounds like a bigger change.

Anyway, as you say, it's probably unlikely anything will actually update the state after the user has started typing, at least the way the rest of the app works right now (with no push updates from the server, no background calls, or anything like that), so we should be fine for now.